pkg/image/controller/factory.go

```diff
@@ -1,59 +0,0 @@
-package controller
-
-import (
-	"time"
-
-	kapi "k8s.io/kubernetes/pkg/api"
-	"k8s.io/kubernetes/pkg/client/cache"
-	"k8s.io/kubernetes/pkg/fields"
-	"k8s.io/kubernetes/pkg/labels"
-	"k8s.io/kubernetes/pkg/runtime"
-	"k8s.io/kubernetes/pkg/util"
-	kutil "k8s.io/kubernetes/pkg/util"
-	"k8s.io/kubernetes/pkg/watch"
-
-	"github.com/openshift/origin/pkg/client"
-	"github.com/openshift/origin/pkg/controller"
-	"github.com/openshift/origin/pkg/image/api"
-)
-
-// ImportControllerFactory can create an ImportController.
-type ImportControllerFactory struct {
-	Client client.Interface
-}
-
-// Create creates an ImportController.
-func (f *ImportControllerFactory) Create() controller.RunnableController {
-	lw := &cache.ListWatch{
-		ListFunc: func() (runtime.Object, error) {
-			return f.Client.ImageStreams(kapi.NamespaceAll).List(labels.Everything(), fields.Everything())
-		},
-		WatchFunc: func(resourceVersion string) (watch.Interface, error) {
-			return f.Client.ImageStreams(kapi.NamespaceAll).Watch(labels.Everything(), fields.Everything(), resourceVersion)
-		},
-	}
-	q := cache.NewFIFO(cache.MetaNamespaceKeyFunc)
-	cache.NewReflector(lw, &api.ImageStream{}, q, 2*time.Minute).Run()
-
-	c := &ImportController{
-		streams:  f.Client,
-		mappings: f.Client,
-	}
-
-	return &controller.RetryController{
-		Queue: q,
-		RetryManager: controller.NewQueueRetryManager(
-			q,
-			cache.MetaNamespaceKeyFunc,
-			func(obj interface{}, err error, retries controller.Retry) bool {
-				util.HandleError(err)
-				return retries.Count < 5
-			},
-			kutil.NewTokenBucketRateLimiter(1, 10),
```

Comment:
> Note that I'm not against termination in a large number of
> deterministic stop conditions - 404, access denied, unrecognized
> non-connection errors.  I'm just highlighting that certain errors
> beyond conflict should be retried because transient failures are
> inevitable.  The new import endpoint will allow that distinction to be
> drawn for partial completion, but we still need to make an effort.

You said you're refactoring `Next`.  I'd expect that you're handling those.  If not, would you like me to the do the refactor here?  If we're doing a large import, it makes sense to have the dockerclient retry near the point of failure to avoid rework.

On the openshift resource mutation side, the update problem exists even without this pull, so you'll have to have patch conflicts, compatibility, and coverage evaluation to handle update conflicts at the point of failure anyway.  Again, is that in your refactor or do you want for this one?

The only thing this is changing is when the retry happens.  Now it happens on a sync-period instead of immediately at time of failure.